File: src/main/java/pl/rejsykoalicja/charter/dto/PayoffDto.java
package pl.rejsykoalicja.charter.dto;

import lombok.Builder;
import lombok.Data;
import pl.rejsykoalicja.charter.enums.Discount;

import java.math.BigDecimal;
import java.util.List;

@Data
@Builder
public class PayoffDto {
    private BigDecimal totalCost;
    private Integer discountValue;
    private List<Discount> discounts;
    private VoucherDto voucher;
    private List<PaymentDto> payments;
}

File: client/rejsy-koalicja/src/shared/models/payoffDto.ts
import {Discount} from "../enums/discount";
import {VoucherDto} from "./voucherDto";
import {PaymentDto} from "./paymentDto";

export interface PayoffDto {
    totalCost?: number
    discountValue?: number
    discounts?: Discount[]
    voucher?: VoucherDto
    payments?: PaymentDto[]
}
